lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -189,3 +189,38 @@ function _gutenberg_get_global_stylesheet_clean_cache_upon_upgrading( $upgrader,
 	}
 }
 
+/*
+ * Function to get the settings resulting of merging core, theme, and user data.
+ *
+ * @param array $path    Path to the specific setting to retrieve. Optional.
+ *                       If empty, will return all settings.
+ * @param array $context {
+ *     Metadata to know where to retrieve the $path from. Optional.
+ *
+ *     @type string $block_name Which block to retrieve the settings from.
+ *                              If empty, it'll return the settings for the global context.
+ *     @type string $origin     Which origin to take data from.
+ *                              Valid values are 'all' (core, theme, and user) or 'base' (core and theme).
+ *                              If empty or unknown, 'all' is used.
+ * }
+ *
+ * @return array The settings to retrieve.
+ */
+function gutenberg_get_global_settings( $path = array(), $context = array() ) {
+	if ( ! empty( $context['block_name'] ) ) {
+		$path = array_merge( array( 'blocks', $context['block_name'] ), $path );
+	}
+
+	$origin = 'custom';
+	if ( ! wp_theme_has_theme_json() ) {
+		// For themes with no theme.json skip querying the database for user data (custom origin).
+		$origin = 'theme';
+	} elseif ( isset( $context['origin'] ) && 'base' === $context['origin'] ) {
+		$origin = 'theme';
+	} elseif ( isset( $context['origin'] ) && 'all' === $context['all'] ) {
+		$origin = 'custom';
+	}
```

Comment:
This condition is unnecessary since it's set to `custom` initially in line 214 anyway. Also, is there a typo in `$context['all']`, shouldn't this be `$context['origin']`?

If the primary reason for this logic here is to make the intention more clear, it would be better to instead add a comment above line 214 saying that this should be used for the `all` origin context or something like that.